ui/src/pages/version/components/forms/components/transformer/components/table_inputs/OrdinalEncoderMapper.js

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import { EuiButtonIcon, EuiFieldText, EuiFormRow } from "@elastic/eui";
+import { InMemoryTableForm, useOnChangeHandler } from "@gojek/mlp-ui";
+
+export const OrdinalEncoderMapper = ({
+  mappings,
+  onChangeHandler,
+  errors = {}
+}) => {
+  const { onChange } = useOnChangeHandler(onChangeHandler);
+
+  const [items, setItems] = useState([
+    ...Object.keys(mappings).map((v, idx) => ({
+      idx,
+      original: v,
+      transformed: mappings[v]
+    })),
+    { idx: Object.keys(mappings).length }
+  ]);
+
+  useEffect(
+    () => {
+      let newMappings = {};
+      items
+        .filter(item => item.original && item.original !== "")
+        .forEach(item => {
+          newMappings[item.original] = item.transformed;
+        });
+      onChange("mapping")(newMappings);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [items]
+  );
+
+  const onDeleteItem = idx => () => {
+    items.splice(idx, 1);
+    setItems([...items.map((v, idx) => ({ ...v, idx }))]);
+  };
+
+  const getRowProps = item => {
+    const { idx } = item;
+    const isInvalid = !!errors[idx];
+    return {
+      className: isInvalid ? "euiTableRow--isInvalid" : "",
+      "data-test-subj": `row-${idx}`
+    };
+  };
+
+  const onChangeRow = (idx, field) => {
+    return e => {
+      items[idx] = { ...items[idx], [field]: e.target.value };
+
+      setItems(_ =>
+        field === "original" &&
```

Comment:
this function is called when original and transformed fields are changed, why you only check the `original` field here?

Reply:
same thing, this is from RenameColumn.js.
But I think it is checking only the original because, this is the part where it will set the key-value pair. If the original is not set, it don't have the key to create the key-value pair. Once the key is set, any changes to key or value will be reflected immediately on the yaml.